bundle/edu.gemini.pot/src/test/scala/edu/gemini/spModel/target/env/GuideGroupSpec.scala

```diff
@@ -177,90 +175,92 @@ class GuideGroupSpec extends Specification with ScalaCheck with Arbitraries {
       }
 
     "return matching guide probe targets for all guide probes with associated targets" in
-      forAll { (g: GuideGroup) =>
+      forAll { g: GuideGroup =>
         val gpts1 = g.getAll.asScalaList
         val gpts2 = AllProbes.flatMap { gp => g.get(gp).asScalaOpt }
         gpts1 == gpts2
       }
   }
-/*
-  "GuideGroup getAll" should {
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup putAll" should {
+    "contain the original primary guide probes and all primary guide probes for which targets have been added" in
+      forAll { (g: GuideGroup, gps: Set[GuideProbe]) =>
+        val gpts = gps.map(GuideProbeTargets.create(_, new SPTarget())).toList.asImList
+        val allGps = g.getPrimaryReferencedGuiders.asScala.toSet ++ gps
+        g.putAll(gpts).getPrimaryReferencedGuiders.asScala.toSet == allGps
```

Comment:
This seems like a good property to test but we should probably add something like:

* behave as if individual calls to `put` were made in sequence for each `GuideProbeTargets` instance

without something like this you can meet the spec by adding any random `SPTarget`, not just the ones provided in the argument to `putAll`.